test_scripts/Capabilities/PersistingHMICapabilities/common.lua

```diff
@@ -556,19 +586,43 @@ function m.registerAppsSuspend( pCapResponse, pHMIParams )
   local mobSession2 = actions.mobile.getSession(appSessionId2)
   local mobSession3 = actions.mobile.getSession(appSessionId3)
 
-  actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered",
-    { application = { appName = actions.app.getParams(appSessionId1).appName }},
-    { application = { appName = actions.app.getParams(appSessionId2).appName }},
-    { application = { appName = actions.app.getParams(appSessionId3).appName }})
-  :Do(function(exp, data)
-      if exp.occurences == 1 then
-        actions.app.setHMIId(data.params.application.appID, appSessionId1)
-      elseif exp.occurences == 2 then
-        actions.app.setHMIId(data.params.application.appID, appSessionId2)
-      else
-        actions.app.setHMIId(data.params.application.appID, appSessionId3)
-      end
-    end)
+  local isAppSessionId = {
+    [1] = false,
+    [2] = false,
+    [3] = false
+  }
+  local function validationBCOnAppRegistered(pAppId)
+    if isAppSessionId[pAppId] == false then
+      isAppSessionId[pAppId] = true
+    else
+      actions.run.fail("SDL sends BasicCommunication.OnAppRegistered notification twice for app" .. pAppId)
+    end
+  end
+  actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
+  :ValidIf(function(_,data)
+    local appName = data.params.application.appName
+    if appName == actions.app.getParams(appSessionId1).appName then
+      validationBCOnAppRegistered(appSessionId1)
+    elseif appName == actions.app.getParams(appSessionId2).appName then
+      validationBCOnAppRegistered(appSessionId2)
+    elseif appName == actions.app.getParams(appSessionId3).appName then
+      validationBCOnAppRegistered(appSessionId3)
+    else
+      actions.run.fail("SDL sends unexpected BasicCommunication.OnAppRegistered notification for app" .. appName)
+    end
+    return true
+  end)
```

Comment:
@VjKlepikov I would propose to simplify this validation:
```suggestion
local isAppRegisteredReceived = {
  [actions.app.getParams(appSessionId1).appName] = false,
  [actions.app.getParams(appSessionId2).appName] = false,
  [actions.app.getParams(appSessionId3).appName] = false
}

local function validateBCOnAppRegistered(pAppName)
  if isAppRegisteredReceived[pAppName] == false then
    isAppRegisteredReceived[pAppName] = true
    return true
  elseif isAppRegisteredReceived[pAppName] == nil then
    return false, "SDL sends unexpected BC.OnAppRegistered notification for app with name " .. pAppName
  else
    return false, "SDL sends BC.OnAppRegistered notification twice for app with name" .. pAppName
  end
end

actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
:ValidIf(function(_,data)
  return validateBCOnAppRegistered(data.params.application.appName)
end)
```
=>
```suggestion
local isAppRegisteredReceived = {
  [actions.app.getParams(appSessionId1).appName] = false,
  [actions.app.getParams(appSessionId2).appName] = false,
  [actions.app.getParams(appSessionId3).appName] = false
}

actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered")
:ValidIf(function(_, data)
  local appName = data.params.application.appName
  if isAppRegisteredReceived[appName] == false then
    isAppRegisteredReceived[appName] = true
    return true
  elseif isAppRegisteredReceived[appName] == nil then
    return false, "SDL sends unexpected BasicCommunication.OnAppRegistered notification for app with name " .. appName
  else
    return false, "SDL sends BasicCommunication.OnAppRegistered notification twice for app with name" .. appName
  end
end)
```

Reply:
Fixed (see 3ab319)